octopoes/ddl/schemas/openkat_schema.graphql

```diff
@@ -0,0 +1,161 @@
+# Native Kat
+directive @constraint (
+    format: String = ""
+) on FIELD_DEFINITION
+
+directive @natural_key (
+    fields: [String]!
+) on FIELD_DEFINITION
+
+directive @format (
+    format: String = ""
+) on FIELD_DEFINITION
+
+directive @reverse_name (
+    name: String = ""
+) on FIELD_DEFINITION
+
+interface BaseObject {
+   primary_key: ID!
+}
+
+interface OOI implements BaseObject {
+   primary_key: ID!
+   human_readable: String!
+}
+
+
+# Networking
```

Comment:
I was wondering if there were any plans or ideas to eventually generate these?

Reply:
Perhaps. In general, GraphQL schemas are optimized for reading, not writing. Therefor, types implementing interfaces need to 'redeclare' the inherited types. This is clarify the type in conflicting types in e.g. diamond inheritance.

There are quite a lot of discussions about this. E.g. https://stackoverflow.com/questions/43449564/why-graphql-implements-need-to-duplicate-the-fields-is-that-mandatory-if-yes

Now that I think about it, perhaps we could use `__typename`? But it might create a discrepancy between the JSON and graphql representation of an object.. :-\